v4/export/config.go

```diff
@@ -28,28 +33,85 @@ func DefaultConfig() *Config {
 		Logger:        &DummyLogger{},
 		FileSize:      UnspecifiedSize,
 		OutputDirPath: ".",
+		ServerInfo:    UnknownServerInfo,
+		SortByPk:      false,
 	}
 }
 
 func (conf *Config) getDSN(db string) string {
 	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", conf.User, conf.Password, conf.Host, conf.Port, db)
 }
 
-func extractOutputConfig(conf *Config) *Config {
-	return &Config{
-		Logger:        conf.Logger,
-		FileSize:      conf.FileSize,
-		OutputDirPath: conf.OutputDirPath,
+const UnspecifiedSize = 0
+
+type ServerInfo struct {
+	ServerType    ServerType
+	ServerVersion *ServerVersion
+}
+
+var UnknownServerInfo = ServerInfo{
+	ServerType:    UnknownServerType,
+	ServerVersion: nil,
+}
+
+var versionRegex = regexp.MustCompile("^(\\d+\\.){2}\\d+")
+var tidbVersionRegex = regexp.MustCompile("v(\\d+\\.){2}\\d+")
```

Comment:
For TiDB, if the server reports `v3.1.0-beta.1`, that `-beta.1` part is also important I think.